subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -192,6 +192,12 @@ Gradle 8.x supports AGP 7.3 and above if you configure the following property:
 android.experimental.legacyTransform.forceNonIncremental=true
 ```
 
+==== Changes to Artifact Transform selection algorithm
+
+Prior to Gradle 8.0, the artifact transform selection algorithm could potentially have selected inefficient transforms, where an intermediate
+step in the transformation chain could have been satisfied by an existing variant. Now, the selection algorithm has been optimized
+to always prefer transform chains of the fewest steps, while still enforcing the same transformation attribute matching principles as before.
```

Comment:
```suggestion
==== Optimized Artifact Transform selection algorithm

Prior to Gradle 8.0, the artifact transform selection algorithm sometimes selected suboptimal transforms.
In certain situations, an intermediate step in the transformation chain could have been satisfied by an existing variant, but Gradle did not use the existing variant.
Now, an optimized selection algorithm uses transform chains containing the smallest number of steps.
```
Suggestions:

- clarify title
- break up the very long, hard to read first sentence into two more readable sentences
- switch "inefficient" to "suboptimal" because it doesn't sound like this was an egregious inefficiency -- just a missed optimization
- omit final clause of final sentence -- I think this goes without saying, since violating those principles would be breaking behavior


I would consider eliminating this paragraph entirely. It feels more like it belongs in release notes than the "upgrading" guide, since as far as I can tell users never have to think about this: it's a (nice!) performance improvement "under the hood". We should probably reserve the "upgrading" guides for user actionable changes.

Reply:
Thanks for the tips! I agree that this information is not actionable by users and therefore doesn't really belong in the upgrade guide. Since there are also no new features, I've gone with your suggestion to remove it entirely instead of adding it to the release notes.